sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresInt4Tests.java

```diff
@@ -0,0 +1,419 @@
+                        " true \n" +
+                        "------\n" +
+                        " t\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT '2'::INT2 * '2'::INT4 = '16'::INT4 / '4'::INT2 AS true;\n" +
+                        " true \n" +
+                        "------\n" +
+                        " t\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT '1000'::INT4 < '999'::INT4 AS false;\n" +
+                        " false \n" +
+                        "-------\n" +
+                        " f\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 + 1 AS ten;\n" +
+                        " ten \n" +
+                        "-----\n" +
+                        "  10\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT 2 + 2 / 2 AS three;\n" +
+                        " three \n" +
+                        "-------\n" +
+                        "     3\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT (2 + 2) / 2 AS two;\n" +
+                        " two \n" +
+                        "-----\n" +
+                        "   2\n" +
+                        "(1 row)"
+        );
+    }
+
+    // Check PostgresInt2Tests::testSelectOverflow for details
+    @Test @Ignore
+    public void testSelectOverflow() {
+        String error = "INT4 out of range";
+
+        this.shouldFail("SELECT i.f1, i.f1 * '2'::INT4 AS x FROM INT4_TBL i", error);
+        this.shouldFail("SELECT i.f1, i.f1 * '2'::INT4 AS x FROM INT4_TBL i", error);
+        this.shouldFail("SELECT i.f1, i.f1 + '2'::INT2 AS x FROM INT4_TBL i", error);
+        this.shouldFail("SELECT i.f1, i.f1 + '2'::INT4 AS x FROM INT4_TBL i", error);
+        this.shouldFail("SELECT i.f1, i.f1 - '2'::INT2 AS x FROM INT4_TBL i", error);
+        this.shouldFail("SELECT i.f1, i.f1 - '2'::INT4 AS x FROM INT4_TBL i", error);
+    }
+
+    @Test @Ignore("shift operator is not supported")
+    public void testLeftShift() {
+        this.qs(
+                "SELECT (-1::int4<<31)::text;\n" +
+                        "    text     \n" +
+                        "-------------\n" +
+                        " -2147483648\n" +
+                        "(1 row)\n" +
+                        "\n" +
+                        "SELECT ((-1::int4<<31)+1)::text;\n" +
+                        "    text     \n" +
+                        "-------------\n" +
+                        " -2147483647\n" +
+                        "(1 row)"
+        );
+    }
+
+    // This passes for the Calcite version but fails for the run time version
+    // Check PostgresInt2Tests::testINT2MINOverflow for details
```

Comment:
this is clearly a bug on our side, please file an issue and add it to the Ignore tag